modin/pandas/base.py

```diff
@@ -1278,11 +1280,18 @@ def diff(self, periods=1, axis=0):  # noqa: PR01, RT01, D200
         """
         First discrete difference of element.
         """
+        # Attempting to match pandas error behavior here
+        if not isinstance(periods, int):
+            raise ValueError(f"periods must be an int. got {type(periods)} instead")
+
+        # Attempting to match pandas error behavior here
+        for dtype in self._get_dtypes():
+            if not is_numeric_dtype(dtype):
+                raise TypeError(f"unsupported operand type for -: got {dtype}")
+
         axis = self._get_axis_number(axis)
         return self.__constructor__(
-            query_compiler=self._query_compiler.diff(
-                fold_axis=axis, axis=axis, periods=periods
-            )
+            query_compiler=self._query_compiler.diff(axis=axis, periods=periods)
```

Comment:
Nit: For consistency, I like to keep the same argument order as the function signature even though this is not necessary for functionality.

```suggestion
            query_compiler=self._query_compiler.diff(periods=periods, axis=axis)
```